crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -37,13 +37,74 @@ cfg_if::cfg_if! {
+            }
+            pub fn decommit_memory_pages(backend: PoolingBackend, addr: *mut u8, len: usize) -> Result<()> {
+                match backend {
+                    PoolingBackend::Mmap => super::linux::decommit_memory_pages(backend, addr, len),
+                    #[cfg(feature = "memfd-allocator")]
+                    PoolingBackend::MemFd => super::linux::decommit_memory_pages(backend, addr, len),
+                    #[cfg(feature = "uffd")]
+                    PoolingBackend::Uffd => super::uffd::decommit_memory_pages(backend, addr, len),
+                }
+            }
+            pub fn commit_table_pages(backend: PoolingBackend, addr: *mut u8, len: usize) -> Result<()> {
+                match backend {
+                    PoolingBackend::Mmap => super::linux::commit_table_pages(backend, addr, len),
+                    #[cfg(feature = "memfd-allocator")]
+                    PoolingBackend::MemFd => super::linux::commit_table_pages(backend, addr, len),
+                    #[cfg(feature = "uffd")]
+                    PoolingBackend::Uffd => super::uffd::commit_table_pages(backend, addr, len),
+                }
+            }
+            pub fn decommit_table_pages(backend: PoolingBackend, addr: *mut u8, len: usize) -> Result<()> {
+                match backend {
+                    PoolingBackend::Mmap => super::linux::decommit_table_pages(backend, addr, len),
+                    #[cfg(feature = "memfd-allocator")]
+                    PoolingBackend::MemFd => super::linux::decommit_table_pages(backend, addr, len),
+                    #[cfg(feature = "uffd")]
+                    PoolingBackend::Uffd => super::uffd::decommit_table_pages(backend, addr, len),
+                }
+            }
+
+            #[cfg(feature = "async")]
+            pub fn commit_stack_pages(backend: PoolingBackend, addr: *mut u8, len: usize) -> Result<()> {
+                match backend {
+                    PoolingBackend::Mmap => super::linux::commit_stack_pages(backend, addr, len),
+                    #[cfg(feature = "memfd-allocator")]
+                    PoolingBackend::MemFd => super::linux::commit_stack_pages(backend, addr, len),
+                    #[cfg(feature = "uffd")]
+                    PoolingBackend::Uffd => super::uffd::commit_stack_pages(backend, addr, len),
+                }
+            }
+
+            #[cfg(feature = "async")]
+            pub fn decommit_stack_pages(backend: PoolingBackend, addr: *mut u8, len: usize) -> Result<()> {
+                match backend {
+                    PoolingBackend::Mmap => super::linux::decommit_stack_pages(backend, addr, len),
+                    #[cfg(feature = "memfd-allocator")]
+                    PoolingBackend::MemFd => super::linux::decommit_stack_pages(backend, addr, len),
+                    #[cfg(feature = "uffd")]
+                    PoolingBackend::Uffd => super::uffd::decommit_stack_pages(backend, addr, len),
+                }
+            }
```

Comment:
I agree that the need to do something like this is definitely overkill, althought the suggestion below of impl-as-methods-on-something might cut down on the wordiness here.